tensor/datafs/data.go

```diff
@@ -0,0 +1,314 @@
+// Values are initialized to zero value for type.
+// All names must be unique in the directory.
+// Returns the first item created, for immediate use of one value.
+func New[T any](dir *Data, names ...string) (*Data, error) {
+	if len(names) == 0 {
+		err := errors.New("datafs.New requires at least 1 name")
+		return nil, err
+	}
+	var r *Data
+	var errs []error
+	for _, nm := range names {
+		var v T
+		d, err := NewData(dir, nm)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		d.Value = v
+		if r == nil {
+			r = d
+		}
+	}
+	return r, errors.Join(errs...)
+}
+
+// NewTensor returns a new Tensor of given data type, shape sizes,
+// and optional dimension names, in given directory Data item.
+// name must be unique in the directory.
+func NewTensor[T string | bool | float32 | float64 | int | int32 | byte](dir *Data, name string, sizes []int, names ...string) (tensor.Tensor, error) {
+	tsr := tensor.New[T](sizes, names...)
+	d, err := NewData(dir, name)
+	d.Value = tsr
+	return tsr, err
+}
+
+// NewTable makes new table.Table(s) in given directory,
+// for given name(s) (at least one is required).
+// All names must be unique in the directory.
+// Returns the first table created, for immediate use of one item.
+func NewTable(dir *Data, names ...string) (*table.Table, error) {
+	if len(names) == 0 {
+		err := errors.New("datafs.New requires at least 1 name")
+		return nil, err
+	}
+	var r *table.Table
+	var errs []error
+	for _, nm := range names {
+		t := table.NewTable(nm)
+		d, err := NewData(dir, nm)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		d.Value = t
+		if r == nil {
+			r = t
+		}
+	}
+	return r, errors.Join(errs...)
+}
+
+///////////////////////////////
+// Data Access
+
+// IsNumeric returns true if the [DataType] is a basic scalar
+// numerical value, e.g., float32, int, etc.
+func (d *Data) IsNumeric() bool {
+	return reflectx.KindIsNumber(d.DataType())
+}
+
+// DataType returns the type of the data elements in the tensor.
+// Bool is returned for the Bits tensor type.
+func (d *Data) DataType() reflect.Kind {
+	if d.Value == nil {
+		return reflect.Invalid
+	}
+	return reflect.TypeOf(d.Value).Kind()
+}
+
+func (d *Data) KnownFileInfo() fileinfo.Known {
+	if tsr := d.AsTensor(); tsr != nil {
+		return fileinfo.Tensor
+	}
+	kind := d.DataType()
+	if reflectx.KindIsNumber(kind) {
+		return fileinfo.Number
+	}
+	if kind == reflect.String {
+		return fileinfo.String
+	}
+	return fileinfo.Unknown
+}
+
+// AsTensor returns the data as a tensor if it is one, else nil.
+func (d *Data) AsTensor() tensor.Tensor {
+	tsr, _ := d.Value.(tensor.Tensor)
+	return tsr
+}
+
+// AsTable returns the data as a table if it is one, else nil.
+func (d *Data) AsTable() *table.Table {
+	dt, _ := d.Value.(*table.Table)
+	return dt
+}
+
+// AsFloat64 returns data as a float64 if it is a scalar value
+// that can be so converted.  Returns false if not.
+func (d *Data) AsFloat64() (float64, bool) {
```

Comment:
We need all of these `reflectx`-esque converter functions?

Reply:
this is the primary api for scalars. yes.